### ISSUE TITLE
How to completey remove Firefox Multi-Account Containers Extension?

### ISSUE THREAD

**Issue by DonaldLuck117** [mozilla/multi-account-containers#2704]
### Before submitting a bug report

- [X] I updated to the latest version of Multi-Account Container and tested if I can reproduce the issue
- [X] I searched for existing reports to see if it hasn't already been reported

### Step to reproduce

1. I disabled all the syncing options. No sign-ins
2. I hit the “Remove” button under “Manage Your Extensions” / Homepage
3. I restarted Firefox
4. “Open New Container Tab” in right-click context menu still works

### Actual behavior

The Multi-Account Container Extension cannot be 100% completely removed / deleted / uninstalled

### Expected behavior

Is there not a way to completely remove the extension? The context menu option “Open New Container Tab” still shows up. It still works which means it’s still running, even though I removed it from the extensions

### Additional informations

1) Deleting the containers.json file in the profile folder after exiting Firefox did not remove it because Containers.json file gets recreated after opening Firefox

2) Changing the value to false for privacy.userContext.enabled, privacy.userContext.ui.enabled in about:config does not work. These options reset

3) Removing these lines in prefs.js file is the only way to _hide_ “Open New Container Tab”:
```
user_pref("privacy.userContext.enabled", true);
user_pref("privacy.userContext.ui.enabled", true);
```

However, I’d really like to remove the extension, not hide it. What’s the ID for this extension? Are there registries associated with it? I just want to completely get rid of it. That’s all I ask…

**Comment by dotproto** [comment]
It's a bit confusing, but the container-related UI elements that you're seeing after removing the extension come from Firefox's containers feature, not (directly) from the Firefox Multi-Account Containers extension. Installing this extension (or any other with the "contextualIdentities" permission) enables the containers feature in Firefox. Removing the extension _does not_ disable the Firefox feature.

The `privacy.userContext.enabled` and `privacy.userContext.ui.enabled` prefs entries control the Firefox feature. The first preference controls whether or not Firefox containers are enabled. The second preference controls whether or not the "Enable Container Tabs" setting appears in the Tabs section of the `about:settings` page.

The easiest way to hide containers-related features in Firefox's UI is to navigate to `about:settings` and uncheck the "Enable Container Tabs" option. The more complicated (and possibly error-prone) way to do it is to remove the two preferences you identified either using `about:config` or by modifying your prefs file. Personally, I'd strongly recommend that most users use the settings page.

**Comment by knghtbrd** [comment]
Yup, @dotproto nailed it. Containers themselves are just part of Gecko-based browsers and can be used without this extension. There are even some alternatives to this extension to control the browser feature, once in awhile with better results even.

Container Proxy, for example, allows you to define multiple proxies and assign them to one or more containers elegantly. You can assign a proxy to a container with multi-account-containers, but the UI, by comparison, is a gorilla holding the proxy config against the side of the container and hitting it with a rock until it adheres itself to the container.

Under Settings → General → Tabs you'll see the ability to turn off containers in the browser if you want to do so, or there'll be something like "tabContainer requires Container Tabs." there telling you that some container is using the feature and it cannot be turned off.

Basically, this is not a bug.
